extensions/resteasy-reactive/rest-client-reactive/deployment/src/test/java/io/quarkus/rest/client/reactive/error/BlockingExceptionMapperTest.java

```diff
@@ -0,0 +1,163 @@
+package io.quarkus.rest.client.reactive.error;
+
+import static io.quarkus.rest.client.reactive.RestClientTestUtil.setUrlForClass;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.io.InputStream;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.UriBuilder;
+
+import org.eclipse.microprofile.rest.client.RestClientBuilder;
+import org.eclipse.microprofile.rest.client.annotation.RegisterProvider;
+import org.eclipse.microprofile.rest.client.ext.ResponseExceptionMapper;
+import org.eclipse.microprofile.rest.client.inject.RegisterRestClient;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.jboss.resteasy.reactive.common.core.BlockingNotAllowedException;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.rest.client.reactive.ClientExceptionMapper;
+import io.quarkus.test.QuarkusUnitTest;
+import io.smallrye.common.annotation.Blocking;
+import io.vertx.core.Context;
+
+public class BlockingExceptionMapperTest {
```

Comment:
Can we add a test where both types of exception mappers are used for the same client?

Reply:
Both exception mappers will be executed on the worker thread because before the error handler is executed, we apply the ClientBlockingResponseHandler.

Reply:
My point is this: Because of [this](https://github.com/quarkusio/quarkus/blob/ebee0a4b66d1b4dba9c51165e78d22fbf3b90d3e/extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/runtime/MicroProfileRestClientResponseFilter.java#L31), I want to have a test that verifies that in the presence of multiple types of mappers (i..e blocking and non-blocking), where of course only one ends up handling the response, each mapper is called on the proper thread

Reply:
Unfortunately, when having a blocking exception mapper, it will use the worker thread even if it's not used. 
I know this is not ideal, but the handlers chain is determined when building the client, though [this line](https://github.com/quarkusio/quarkus/blob/ebee0a4b66d1b4dba9c51165e78d22fbf3b90d3e/extensions/resteasy-reactive/rest-client-reactive/runtime/src/main/java/io/quarkus/rest/client/reactive/runtime/MicroProfileRestClientResponseFilter.java#L31) is executed at request time. 

Do you want something in mind to overcome this situation?

Reply:
I do not  have any specific fixes in mind, but my point is that the use of `@Blocking` on an exception handler should only cause that piece of code to execute on the worker thread, nothing else.

This is one of the reasons I said initially that this is not an easy issue to fix.

Reply:
I changed the solution to support this.